src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -0,0 +1,296 @@
+/-
+Copyright (c) 2021 Hunter Monroe. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hunter Monroe, Kyle Miller, Alena Gusakov
+-/
+import combinatorics.simple_graph.basic
+import data.set.finite
+
+/-!
+# Subgraphs of a simple graph
+
+A subgraph of a simple graph consists of subsets of the graph's vertices and edges such that the
+endpoints of each edge are present in the vertex subset.  The edge subset is formalized as a
+sub-relation of the adjacency relation of the simple graph.
+
+## Main definitions
+
+* `subgraph G` is the type of subgraphs of a `G : simple_graph`
+
+* `subgraph.neighbor_set`, `subgraph.incidence_set`, and `subgraph.degree` are like their
+  `simple_graph` counterparts, but they refer to vertices from `G` to avoid subtype coercions.
+
+* `subgraph.coe` is the coercion from a `G' : subgraph G` to a `simple_graph G'.verts`.
```

Comment:
Oh, this comment is very outdated.  You can't write a `has_coe` instance for the simple reason that there's nothing you can put in place of the underscore in `has_coe (subgraph G) (simple_type _)` since it depends on the subgraph.

In an older version of the graph library, the graph structure was given by a typeclass on the graph term itself, and the issue was that you had to specify all the universe variables yourself because typeclass inference doesn't want to unnecessarily equate universe variables.  You could write the `has_coe` instance, but it was no fun to use.